pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BaseTableUpsertMetadataManager.java

```diff
@@ -81,6 +81,8 @@ public void init(TableConfig tableConfig, Schema schema, TableDataManager tableD
         .setPartialUpsertHandler(partialUpsertHandler).setEnableSnapshot(enableSnapshot).setEnablePreload(enablePreload)
         .setMetadataTTL(metadataTTL).setDeletedKeysTTL(deletedKeysTTL).setConsistencyMode(_consistencyMode)
         .setUpsertViewRefreshIntervalMs(upsertViewRefreshIntervalMs).setTableIndexDir(tableIndexDir)
+        .setDropOutOfOrderRecord(upsertConfig.isDropOutOfOrderRecord())
+        .setEnableConsistentDeletes(upsertConfig.isEnableConsistentDeletes())
```

Comment:
Why is there a separate config for this? Shouldn't we use the new metadata manager whenever deletions are enabled?

Reply:
So folks can enable `deletedKeysTTL` without enabling compaction as well. This config is only required when you are enabling deletedKeysTTL + compaction as this entire PR ensures data consistency for this specific scenario.

Reply:
Yes but won't it become non-intuitive for users to put entire `metadataManagerClass` instead of just enabling a boolean flag. Plus IMO we might make a call in future to consolidate this class with the original class and so that will make all existing tables using this `metadataManagerClass` as backward-incompatible. Better to not strictly depend on className to enable this feature.